data-structures/src/main/java/com/baeldung/list/CircularLinkedList.java

```diff
@@ -0,0 +1,102 @@
+package com.baeldung.list;
+
+public class CircularLinkedList {
+
+    Node head = null;
+    Node tail = null;
+
+    public void addNode(int value) {
+
+        Node newNode = new Node(value);
+
+        // If no elements are present, make the newly addNodeed node as head
```

Comment:
A general comment, do we really need all these comments in the code? Most of them are pretty obvious.